File: backend/halfaday/src/main/java/com/xiaobantian/model/Route.java
package com.xiaobantian.model;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "routes")
public class Route {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    private Integer durationHours;
    private String suitableSeasons;
    private String difficulty;
    private String groupSizeNote;
    private String coverImage;

    @OneToMany(mappedBy = "route", cascade = CascadeType.ALL,
               orphanRemoval = true, fetch = FetchType.LAZY)
    @OrderBy("stopOrder ASC")
    private List<RouteStop> stops = new ArrayList<>();

    public Route() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public Integer getDurationHours() { return durationHours; }
    public void setDurationHours(Integer durationHours) { this.durationHours = durationHours; }

    public String getSuitableSeasons() { return suitableSeasons; }
    public void setSuitableSeasons(String suitableSeasons) { this.suitableSeasons = suitableSeasons; }

    public String getDifficulty() { return difficulty; }
    public void setDifficulty(String difficulty) { this.difficulty = difficulty; }

    public String getGroupSizeNote() { return groupSizeNote; }
    public void setGroupSizeNote(String groupSizeNote) { this.groupSizeNote = groupSizeNote; }

    public String getCoverImage() { return coverImage; }
    public void setCoverImage(String coverImage) { this.coverImage = coverImage; }

    public List<RouteStop> getStops() { return stops; }
    public void setStops(List<RouteStop> stops) { this.stops = stops; }
}

File: frontend/my-app/src/types/route.ts
import type { Place } from './place';

export interface RouteSummary {
  id: number;
  name: string;
  description: string | null;
  durationHours: number | null;
  suitableSeasons: string | null;
  difficulty: string | null;
  groupSizeNote: string | null;
  coverImage: string | null;
}

export interface RouteStop {
  id: number;
  route?: RouteSummary;
  place: Place;
  stopOrder: number;
  stayMinutes: number | null;
  note: string | null;
  transportToNext: string | null;
}

export interface Route {
  id: number;
  name: string;
  description: string | null;
  durationHours: number | null;
  suitableSeasons: string | null;
  difficulty: string | null;
  groupSizeNote: string | null;
  coverImage: string | null;
  stops: RouteStop[];
}